tests/ecmp/test_fgnhg.py

```diff
@@ -173,8 +178,94 @@ def partial_ptf_runner(ptfhost, test_case, dst_ip, exp_flow_count, **kwargs):
             qlen=1000,
             log_file=log_file)
 
+def hash_view_cli_verification(duthost, prefix_list):
+    header = ["FG NHG Prefix", "Next Hop", "Hash buckets"]
+    table = []
+    mode = ""
+    compressed_ipv6_prefixes = [] 
+
+    if ":" in prefix_list[0]:
+        mode = "fgnhg_v6" 
+        for ipv6_prefix in prefix_list:
+            subnet = ipv6_prefix.split("/")[1]
+            ipv6_prefix = ipv6_prefix.split("/")[0]
+            ipv6_prefix = unicode(str(ipaddress.IPv6Address(ipv6_prefix)) + "/" + subnet)
+            compressed_ipv6_prefixes.append(ipv6_prefix)
+        prefix_list = compressed_ipv6_prefixes
+    else:
+        mode = "fgnhg_v4" 
+
+    prefix_list = sorted(prefix_list)
+   
+    for prefix in prefix_list:
+        nh_bucket_dict = {}
+        command = 'redis-cli -n 6 HGETALL "FG_ROUTE_TABLE|' + str(prefix) + '"'
+        data = duthost.command(command,module_ignore_errors=True)
+        data = data['stdout'].split('\n')
+
+        pairs = zip(data[::2], data[1::2])
```

Comment:
@kktheballer would be better to have some comments here just to explain this magic

Reply:
Added some comment.. High level it is just some string manipulation to get nexthop ip <--> bucket pairs that are continuously listed throughout the table